setup.cfg

```diff
@@ -42,6 +42,7 @@ install_requires =
     jsonpatch>=1.15
     pyyaml>=5.1  # for parsing CLI equal-delimited options
     importlib_resources>=1.3.0; python_version < "3.9"  # for resources in schema
+    typing_extensions; python_version == "3.7"  # for TypedDict, Literal
```

Comment:
These are strictly just for type checking right, and not runtime dependencies, so they don't _need_ to be added to [`tests/constraints.txt`](https://github.com/scikit-hep/pyhf/blob/97664bcfcef3b4a459a7f700b0703b40aec9b842/tests/constraints.txt), correct?

Reply:
It's needed for runtime checks (as the import is evaluated at runtime and `typing` is missing some stuff in 3.7)

Reply:
Okay then I'll later go in and figure out what is the minimum version required (some super late at night PR work for the future).